examples/games/alien_cake_addict.rs

```diff
@@ -9,9 +10,23 @@ enum GameState {
     GameOver,
 }
 
+#[derive(Resource)]
+struct SpawnTimer(Timer);
+
+fn timer_expired(time: Res<Time>, mut timer: ResMut<SpawnTimer>) -> ShouldRun {
+    timer.0.tick(time.delta());
+    if timer.0.finished() {
+        timer.0.reset();
+        ShouldRun::Yes
+    } else {
+        ShouldRun::No
+    }
```

Comment:
Instead of manually resetting this could be a repeating timer.
```suggestion
    timer.0.tick(time.delta()).just_finished().into()
```

Reply:
Do we actually want the wraparound behavior in this example?

If my game froze for 15 seconds and then resumed, would the cake not respawn three times in rapid succession?

Reply:
I think both of these changes will do the same thing just one is slightly more concise, since `just_finished` will be reset by the next `timer.0.tick(time.delta())`.

Reply:
In that case `just_finished` and `finished` would return true for that tick, and `times_finished_this_tick` would return `3`.

Reply:
I'm updating this to use the repeating timer.

Just wanted to leave a note that there are differences in behavior between a repeating timer and a timer that you manually reset to zero (no matter how much time elapsed).

The manually-reset timer "delays" future spawns while the repeating timer does not. But in the event of a super long duration between frames (i.e. massive hiccup), the repeating timer makes the "next-next" cake seem to spawn early.